hub/06-separate-core-from-devices.md

```diff
@@ -0,0 +1,381 @@
+
+``` json
+{
+  "testorg/admin": {
+    "admin": true,
+    "email": "admin@testorg",
+    "lastUpdated": "2020-05-16T21:26:49.460Z[UTC]",
+    "password": "********",
+    "updatedBy": "root/root"
+  }
+}
+```
+
+Next, generate an RSA key pair:
+
+``` bash
+hzn key create -l 4096 testorg joe@everywhere.com
+```
+
+Then publish the `core` service to the exchange:
+
+``` bash
+hzn exchange service publish --json-file=horizon/service.definition.json -P
+```
+
+Then verify with:
+
+``` bash
+hzn exchange service list
+```
+
+## Create the `device` Service
+
+Then, let's change to the device service directory and edit the device Service Definition file:
+
+``` bash
+cd ../device/horizon
+vi service.definition.json
+```
+
+Set the following properties:
+
+|| Property | Value
+| description | EdgeX Foundry device random service
+| sharable | singleton
+| requiredServices | ```
+      {
+        "url": "com.github.joewxboy.horizon.edgex.core",
+        "org": "testorg",
+        "versionRange": "[1.0.2,INFINITY)",
+        "arch": "amd64"
+      }
+```
+| userInput | ```
+      { "name": "EXPORT_DISTRO_CLIENT_HOST", "label": "", "type": "string", "defaultValue": "export-client" },
+      { "name": "EXPORT_DISTRO_DATA_HOST", "label": "", "type": "string", "defaultValue": "edgex-core-data" },
+      { "name": "EXPORT_DISTRO_CONSUL_HOST", "label": "", "type": "string", "defaultValue": "edgex-config-seed" },
+      { "name": "EXPORT_DISTRO_MQTTS_CERT_FILE", "label": "", "type": "string", "defaultValue": "none" },
+      { "name": "EXPORT_DISTRO_MQTTS_KEY_FILE", "label": "", "type": "string", "defaultValue": "none" },
+      { "name": "LOG_LEVEL", "label": "logging level", "type": "string", "defaultValue": "info" },
+      { "name": "LOGTO", "label": "where to log", "type": "string", "defaultValue": "." }
+```
+| deployment > services > $SERVICE_NAME | ```
+                "binds": [
+                    "db-data:/data/db",
+                    "log-data:/edgex/logs:rw",
+                    "consul-config:/consul/config",
+                    "consul-data:/consul/data",
+                    "/root/res/device-random-config.toml:/res/configuration.toml:ro",
+                    "/root/res/device-random.yaml:/res/device-random.yaml:ro"
+                ],
+                "specific_ports": [
+                    { "HostPort": "49988/tcp", "HostIP": "0.0.0.0" }
+                ],
+                "image": "edgexfoundry/docker-device-random-go:1.0.0",
+                "privileged": true
+```
+
+Then let's navigate back to the `device` subdirectory and validate the Service Definition file:
+
+``` bash
+cd ..
+export SERVICE_NAME="com.github.joewxboy.horizon.edgex.device"
+export SERVICE_VERSION="1.0.2"
+hzn dev service verify
+```
+
+If all goes well and there were no typos, you should see output similar to the following:
+
+```
+New dependency created: url: com.github.joewxboy.horizon.edgex.core, org: testorg, version: [1.0.2,INFINITY), arch: amd64 .
```

Comment:
Is this msg right? I thought this msg only comes out for hzn dev dependency new, which I dont see you doing in these instructions? Maybe you forgot to add instructions for the new dependency.